### DISCUSSION TITLE
self.area.w use in plugins

### DISCUSSION THREAD

**Discussion by pcause** [sxyazi/yazi#1934] [1. Q&A]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

wezterm

### `yazi --debug` output

```Shell
tmux in wezterm on latest nightly and released 3.3
```

### Describe the bug

I've tried a couple of the markdown previewers and none seem to pass the correct width value to the app they invoke for previews.  They all seem to use self.area.w but is that the correct way to get the width of the preview pane?  If not can you add something to the doc or an example?

### Minimal reproducer

install the glow plugin for previewing markdown

### Validations

- [X] I tried the [latest nightly build](https://yazi-rs.github.io/docs/installation#official-binaries), and the issue is still reproducible
- [X] I updated the debug information (`yazi --debug`) input box to the nightly that I tried
- [ ] I can reproduce it after disabling all custom configs/plugins (`mv ~/.config/yazi ~/.config/yazi-backup`)

**Comment by pcause** [comment]
what is the answer to this question?  this ought to be straightforward to answer.

**Reply by AnirudhG07** [reply]
Well according to the docs, this should be the correct way.
What does it look like for you? For me, glow or rich-preview works just fine.

**Comment by pcause** [comment]
when i look at the docs, it seems to give the total length of the line, not just the preview area.   I want yo get the width, in characters, of the preview area only.

**Comment by pcause** [comment]
I don't think so.  According to the docs, this seems to be the line width.  Somehow you have to get the 3rd element of the horizontal layout and then get the width of that

**Comment by sxyazi** [comment]
I checked [the documentation](https://yazi-rs.github.io/docs/plugins/overview#previewer) and found no errors. The variable "area.w" represents the available width of the preview area and is measured in cells rather than characters, because different characters occupy different numbers of cells in the terminal (for example, Latin characters occupy 1 cell while CJK characters occupy 2 cells).

You can find the built-in plugins at https://github.com/sxyazi/yazi/tree/main/yazi-plugin/preset/plugins as examples
